src/python/tests/core/platforms/android/stack_analyzer_test.py

```diff
@@ -0,0 +1,120 @@
+kernel/private/msm-google u'40e9b2ff3a280a8775cfcd5841e530ce78f94355'
+kernel/private/msm-google-extra/audiokernel u'112a618d5b757b0600c69f7385892b3f57ccd93e'
+kernel/private/msm-google-modules/data-kernel u'e7210f09d00c91f87b295c7a952f040c73506cc0'
+kernel/private/msm-google-modules/fts_touch u'8f6a4e9f5649deff59174ffed1d5c2af196d9f63'
+kernel/private/msm-google-modules/qca-wfi-host-cmn u'7d4b05ac12d6a1b5d5247da35ae7e370a2cba07d'
+kernel/private/msm-google-modules/qcacld u'0a077b0073c48555d0edb2b9b0510fb883181828'
+kernel/private/msm-google-modules/wlan-fw-api u'53d899727e4278f4e9fb46328d740a8fb2d9a493'
+kernel/private/tests/patchwork u'204e78fb6d905016bfc16ebe7b64547f388cfdb5'
+kernel/tests u'bfef3bb78b23cb3f3f12a6880ecafd5def3b66a5'
+platform/external/fff u'c82edb1fc60dc81bd319d9b8d0bee9f8963a6960'
+platform/external/googletest u'a037984aea3317260edd1127abb39e30e845bc94'
+platform/prebuilts/clang/host/linux-x86 u'4b1f275e6b3826c86f791ae8c4d5ec3563c2fc11'
+platform/prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9 u'961622e926a1b21382dba4dd9fe0e5fb3ee5ab7c'
+platform/prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9 u'cb7b3ac1b7fdb49474ff68761909934d1142f594'
+platform/prebuilts/misc u'15560bb32cdb9b47db48eb4865b736df9708a8fe'
+platform/tools/repohooks u'233b8010f7f5e3c544b47c68ffae781860156945'
+"""
+_real_read_data_from_file = None
+
+
+# pylint: disable=unused-argument
+def _mock_symbolize_stacktrace(stacktrace, enable_inline_frames=True):
+  """No-op mocked version of symbolize_stacktrace."""
+  return stacktrace
+
+
+def _mock_fetch_artifact_get(bid,
+                             target,
+                             regex,
+                             output_directory,
+                             output_filename_override=None):
+  if output_filename_override:
+    artifact_path = os.path.join(output_directory, output_filename_override)
+    with open(artifact_path, 'w') as artifact_file:
+      artifact_file.write(KERNEL_REPRO)
+
+
+# pylint: enable=unused-argument
+
+
+def _mock_read_data_from_file(file_path, eval_data=True, default=None):
+  if file_path.endswith('repo.prop'):
+    return _real_read_data_from_file(file_path, eval_data, default)
+
+  return None
+
+
+class AndroidStackAnalyzerTest(unittest.TestCase):
+  """Android specific Stack analyzer tests."""
+
+  def setUp(self):
+    test_helpers.patch_environ(self)
+    test_helpers.patch(self, [
+        'crash_analysis.stack_parsing.stack_symbolizer.symbolize_stacktrace',
+        'metrics.logs.log_error',
+    ])
+    os.environ['JOB_NAME'] = TEST_JOB_NAME
+
+    self.mock.symbolize_stacktrace.side_effect = _mock_symbolize_stacktrace
+
+  def _read_test_data(self, name):
+    """Helper function to read test data."""
+    with open(os.path.join(DATA_DIRECTORY, name)) as handle:
+      return handle.read()
+
+  def test_syzkaller_kasan_android_with_env(self):
+    """Test syzkaller kasan."""
+    global _real_read_data_from_file
```

Comment:
Let's avoid this use of global here. We can just save this in self._real_read_data_from_file, and define _mock_read_data_from_file as a member function. You can still do ` self.mock.read_data_from_file.side_effect = self._mock_read_data_from_file` and things will work as expected.

Reply:
+1